lib/ansible/modules/hostname.py

```diff
@@ -170,10 +170,8 @@ def set_permanent_hostname(self, name):
         self.strategy.set_permanent_hostname(name)
 
 
-class GenericStrategy(object):
+class BaseStrategy(object):
```

Comment:
@eqrx Could you please add a changelog entry for this change? Additionally, a unit test to check `BaseStrategy` would be helpful (not a blocker).